.github/workflows/test-all-versions.yml

```diff
@@ -13,8 +13,35 @@ on:
         type: string
 
 jobs:
+  build:
+    strategy:
+      fail-fast: false
+    runs-on: ubuntu-latest
+    env:
+      NPM_CONFIG_UNSAFE_PERM: true
+      NODE_OPTIONS: --max-old-space-size=4096
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 16
+      - name: Install
+        run: npm ci
+      - name: Build
+        run: npm run compile
+      - name: Upload Build Artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: tav-build-cache-${{ github.run_number }}
```

Comment:
note for reviewer: `github.run_number` is unique for the PR so each time the workflow is run this tasks overwrites the artifact saving space. `retention-days` is set to 1 so the artifact will be removed the next day after the last run of this workflow.